haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Tuple, Dict
+
+from dateutil.parser import parse, ParserError
+
+from haystack.nodes.ranker.base import BaseRanker
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
+        weight: float = 0.5,
+        top_k: Optional[int] = None,
+        method: str = "reciprocal_rank_fusion",
+    ):
+        """
+        This Node is used to rerank retrieved documents based on their age. Newer documents will rank higher.
+        The importance of recentness is parametrized through the weight parameter.
+
+        :param date_identifier: Identifier pointing to the date field in the metadata.
+                This is a required parameter, since we need dates for sorting.
+        :param weight: in range [0,1].
+                0 disables sorting by age.
+                0.5 content and age have the same impact
+                1 means sorting only by age, most recent comes first.
+        :param top_k: (optional) How many documents to return.
+                It can make sense to have large top-k values from the initial retrievers and filter docs down in the
+                RecentnessRanker with this top_k parameter.
+        :param method: Method used to combine retriever and recentness. Possible values are 'reciprocal_rank_fusion' (default) and 'score'
+                Make sure to use 'score' method only with retrievers/rankers that give back OK score in range [0,1]
+        """
+
+        super().__init__()
+        self.date_identifier = date_identifier
+        self.weight = weight
+        self.top_k = top_k
+        self.method = method
+
+    # pylint: disable=arguments-differ
+    def run(self, query: str, documents: List[Document], top_k: Optional[int] = None) -> Tuple[Dict, str]:  # type: ignore
```

Comment:
Since this class is inheriting from `BaseRanker` which already provides the methods `run` and `run_batch`, we would normally only try to define the `predict` and `predict_batch` abstract methods.

You can see an example of this in the [cohere ranker code](https://github.com/deepset-ai/haystack/blob/main/haystack/nodes/ranker/cohere.py) and the [sentence transformers ranker code](https://github.com/deepset-ai/haystack/blob/main/haystack/nodes/ranker/sentence_transformers.py). The base class can be seen [here](https://github.com/deepset-ai/haystack/blob/main/haystack/nodes/ranker/base.py) which shows that the `run` and `run_batch` methods are already defined and the unique code for each new ranker is usually defined in `predict` and `predict_batch`. 

I think this should almost be as simple as renaming the `run` and `run_batch` methods here to `predict` and `predict_batch`. You might then need to slightly change what is returned. 

My recommendation would be to make a unit test for this first before making changes so you are confident that the new code changes do not change the results.